### ISSUE TITLE
Neural Network Schematic Diagram

### ISSUE THREAD

**Issue by enchainingrealm** [nianticlabs/monodepth2#164]
I re-implemented MonoDepth2 according to your paper, but my training loss never decreases. It's too easy to miss important implementation details, since they are scattered across various sections of the MonoDepth2 paper, various other papers, and the source code in this repository.

Can you please draw a schematic diagram showing the architecture of the entire MonoDepth2 network in the **monocular video training** case? It should show every computation done in the forward pass, starting from the raw RGB input images and ending with the computed loss. All layer initializations, hyper-parameters, pre-processing and post-processing, etc. should be included in the diagram.

In some cases, the paper and the source code in this repository are different! For example, under the **Multi-scale Estimation** section you state `we first upsample the lower resolution depth maps ... to the input image resolution`. But according to the [source code](https://github.com/nianticlabs/monodepth2/blob/d1c5f03c38305cae4e68917e472d2f9d4eda0b98/trainer.py#L350) you actually upsample the **disparity maps (sigmoid outputs)**, not the depth maps produced by [this function](https://github.com/nianticlabs/monodepth2/blob/d1c5f03c38305cae4e68917e472d2f9d4eda0b98/layers.py#L16).

**Comment by mrharicot** [comment]
Hi,
> Can you please draw a schematic diagram ...

I'm afraid that would be way too much work. We provide the entire code necessary to reproduce the results in the paper, this should be more than enough to allow you to write a faithful reimplementation.

> But according to the source code you actually upsample the disparity maps (sigmoid outputs), not the depth maps produced by this function.

Nice catch! We actually upsample the disparity, I doubt this actually makes a significant difference. If you ever compare the two let us know!
